core/src/mast/mod.rs

```diff
@@ -138,6 +138,11 @@ impl MastForest {
         self.add_node(MastNode::new_dyn())
     }
 
+    /// Adds a dyncall node to the forest, and returns the [`MastNodeId`] associated with it.
+    pub fn add_dyncall(&mut self) -> Result<MastNodeId, MastForestError> {
```

Comment:
`ensure_dyncall` is a method of `MastForestBuilder`, where "ensure" encapsulates the behavior where if an equal node is already present, we'll return that one instead of creating a new one and returning it.

`add_dyncall` is a method of `MastForest`, and adds the node regardless of if an equal one is already present in the `MastForest`.

Reply:
Makes sense, thank you!